crates/uv-fs/src/lib.rs

```diff
@@ -11,6 +11,31 @@ pub mod cachedir;
 mod path;
 pub mod which;
 
+/// Attempt to check if the two paths refer to the same directory.
+pub fn is_same_dir(left: &Path, right: &Path) -> Option<bool> {
+    // First, attempt to check directly
+    if let Ok(value) = same_file::is_same_file(left, right) {
+        return Some(value);
+    };
+
+    // Often, one of the directories won't exist yet so perform the comparison up a level
+    if let (Some(left_parent), Some(right_parent), Some(left_name), Some(right_name)) = (
+        left.parent(),
+        right.parent(),
+        left.file_name(),
+        right.file_name(),
+    ) {
```

Comment:
This is... interesting? Maybe this should be noted in the doc? Like "Unlike `is_same_file`, this will..."

I'm not sure I understand why we should do this extra check for directories but not files.

Reply:
It's more like.. `is_same_file_allow_missing`?

Reply:
(Shit I wrote this code)

If you're going to move it out like this, I think it's important to clarify what's going on. It was narrowly scoped before because it was only intended for that one use case.

Reply:
Sure.